soc/arm/atmel_sam0/common/Kconfig.defconfig.series

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_SAM0
+
+if USB
+
+config HEAP_MEM_POOL_SIZE
```

Comment:
If this is needed for the USB driver, should not this be enabled at the driver's Kconfig?

Reply:
I'm not sure.  The only other example I could find is the STM32 DMA driver, which also enables at the SOC level instead of the driver level.

Reply:
@ioannisg any idea who we can ask to resolve this?

Reply:
This sounds like a continuation of the previous discussion on what should be defconfig-ed at the SoC level.

IMHO, I think it makes sense to use the SoC-level defconfigs to keep track of the recommended SoC-specific default global configurations in a centralised manner, rather than doing `default X if SOC_CONFIG_NAME` all over the repository and making everyone's life harder.

Reply:
Optimistically resolving

Reply:
I think We should create dedicated heap for driver specific things. Similar thing is done for Nordic driver for
`USB_NRFX_EVT_QUEUE_SIZE` and `USB_NRFX_WORK_QUEUE_STACK_SIZE`.  Then the heap size would be picked at driver level.